test/controllers/agent/DeductionsSummaryControllerSpec.scala

```diff
@@ -0,0 +1,124 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.agent
+
+import assets.BaseTestConstants.testAgentAuthRetrievalSuccess
+import assets.CalcBreakdownTestConstants.{calculationDataSuccessModel, calculationDisplaySuccessModel}
+import config.featureswitch.{AgentViewer, DeductionBreakdown, FeatureSwitching}
+import implicits.ImplicitDateFormatterImpl
+import mocks.MockItvcErrorHandler
+import mocks.auth.MockFrontendAuthorisedFunctions
+import mocks.services.{MockCalculationService, MockFinancialTransactionsService, MockIncomeSourceDetailsService}
+import models.calculation.CalcDisplayError
+import play.api.http.Status
+import play.api.mvc.{MessagesControllerComponents, Result}
+import play.api.test.Helpers._
+import testUtils.TestSupport
+import uk.gov.hmrc.http.InternalServerException
+import uk.gov.hmrc.play.language.LanguageUtils
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DeductionsSummaryControllerSpec extends TestSupport with MockCalculationService
+  with MockFrontendAuthorisedFunctions with MockIncomeSourceDetailsService
+  with MockFinancialTransactionsService with FeatureSwitching with MockItvcErrorHandler {
+
+  class Setup {
+
+    val testYear: Int = 2020
+
+    val controller: DeductionsSummaryController = new DeductionsSummaryController(
+      app.injector.instanceOf[views.html.agent.DeductionBreakdown],
+      mockAuthService,
+      mockIncomeSourceDetailsService,
+      mockCalculationService,
+    )(
+      appConfig,
+      app.injector.instanceOf[LanguageUtils],
+      app.injector.instanceOf[MessagesControllerComponents],
+      app.injector.instanceOf[ImplicitDateFormatterImpl],
+      app.injector.instanceOf[ExecutionContext],
+      itvcErrorHandler = mockItvcErrorHandler
+    )
+  }
+
+  "showDeductionsSummary" when {
+    "feature switches AgentViewer and DeductionsBreakdown are enabled" should {
+      "return Status OK and render deductions page" in new Setup {
+
+        enable(AgentViewer)
+        enable(DeductionBreakdown)
+        setupMockAgentAuthRetrievalSuccess(testAgentAuthRetrievalSuccess)
+        mockBothIncomeSources()
+        setupMockGetCalculation("AA111111A", testYear)(calculationDisplaySuccessModel(calculationDataSuccessModel))
+
+        val result: Future[Result] = controller.showDeductionsSummary(taxYear = testYear)(fakeRequestConfirmedClient())
+
+        status(result) shouldBe Status.OK
+      }
+    }
+
+    "feature switches AgentViewer and DeductionsBreakdown are enabled" should {
+      "calcDisplay error" in new Setup {
```

Comment:
Please make test description more readable, explaining the purpose of the test case.